include/oneapi/dpl/pstl/glue_algorithm_impl.h

```diff
@@ -363,13 +363,9 @@ oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy, void>
 replace_if(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __last, _UnaryPredicate __pred,
            const _Tp& __new_value)
 {
-    oneapi::dpl::__internal::__pattern_walk1(
-        ::std::forward<_ExecutionPolicy>(__exec), __first, __last,
-        oneapi::dpl::__internal::__replace_functor<
-            oneapi::dpl::__internal::__ref_or_copy<_ExecutionPolicy, const _Tp>,
-            oneapi::dpl::__internal::__ref_or_copy<_ExecutionPolicy, _UnaryPredicate>>(__new_value, __pred),
-        oneapi::dpl::__internal::__is_vectorization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec),
-        oneapi::dpl::__internal::__is_parallelization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec));
+    auto __dispatch_tag = oneapi::dpl::__internal::__select_backend(::std::forward<_ExecutionPolicy>(__exec), __first);
+    __pattern_replace_if(__dispatch_tag, ::std::forward<_ExecutionPolicy>(__exec), __first, __last, __pred,
```

Comment:
`__pattern_walk1` is going to remain here. Please read the comment near `__pattern_replace_if` for more details.